data/onPostBuild/generateMarkdown.ts

```diff
@@ -0,0 +1,210 @@
+import { GatsbyNode } from 'gatsby';
+import * as path from 'path';
+import * as fs from 'fs';
+import TurndownService from 'turndown';
+import cheerio from 'cheerio';
+import fastGlob from 'fast-glob';
+
+/**
+ * This script generates Markdown static files alongside HTML files for each page.
+ * This allows LLM crawlers and other clients to request markdown versions of pages
+ * which are significantly more token-efficient than HTML.
+ */
+
+const REPORTER_PREFIX = 'generateMarkdown:';
+
+// Constants for content validation
+const REDIRECT_PAGE_MAX_SIZE = 1000; // Maximum size in bytes for redirect pages
+const MIN_CONTENT_LENGTH = 100; // Minimum content length to consider meaningful
+
+// Configure Turndown for documentation-friendly markdown
+const createTurndownService = () => {
+  const turndownService = new TurndownService({
+    headingStyle: 'atx',
+    codeBlockStyle: 'fenced',
+    bulletListMarker: '-',
+    emDelimiter: '_',
+    strongDelimiter: '**',
+    linkStyle: 'inlined',
+    linkReferenceStyle: 'full',
+  });
+
+  // Preserve code block language annotations
+  turndownService.addRule('fencedCodeBlock', {
+    filter: (node) => {
+      return node.nodeName === 'PRE' && node.firstChild && node.firstChild.nodeName === 'CODE';
+    },
+    replacement: (content, node) => {
+      const codeNode = node.firstChild as HTMLElement;
+      const className = codeNode.getAttribute('class') || '';
+      const languageMatch = className.match(/language-(\w+)/);
+      const language = languageMatch ? languageMatch[1] : '';
+
+      // Get the actual code content
+      const code = codeNode.textContent || '';
+
+      return '\n\n```' + language + '\n' + code + '\n```\n\n';
+    },
+  });
+
+  // Remove navigation, headers, footers, and other UI elements
+  turndownService.remove(['nav', 'header', 'footer', 'script', 'style', 'noscript']);
+
+  return turndownService;
+};
+
+// Extract main article content from HTML file
+const extractMainContent = (htmlPath: string): string | null => {
+  try {
+    if (!fs.existsSync(htmlPath)) {
+      return null;
+    }
+
+    const html = fs.readFileSync(htmlPath, 'utf8');
+
+    // Check if this is a redirect page (very small file with window.location.href)
+    if (html.length < REDIRECT_PAGE_MAX_SIZE && html.includes('window.location.href')) {
+      return null; // Skip redirect pages
+    }
+
+    const $ = cheerio.load(html);
+
+    // Remove unwanted elements
+    $('nav, header, footer, script, style, noscript, .sidebar, .navigation').remove();
```

Comment:
I don't think sidebar and navigation exist as selectors to kill, not sure if there was another intention